### PR TITLE
Adds proposal for *-infra task forces

### PR DESCRIPTION
**Pull request by eefahy**
Please check the box next to your name showing that you've read and are good with the proposal:
- [x] @kyledrake 
- [x] @mburns 
- [x] @gmasgras 
- [x] @bigs  
- [x] @jacobheun 
- [x] @mgoelzer
- [ ] @diasdavid - if acceptable, please nominate a dev from js-ipfs and go-ipfs
- [x] @hsanjuan - if acceptable, please nominate a dev from cluster

### CODE REVIEW

**Comment by jacobheun** [comment]
I can't edit the comment, but ✅check.

**Comment by eefahy** [comment]
Yes, I'm using the term "working-group" loosely and decidedly non-standardly but having OKRs and captains is wonderful. I'll add a libp2p doc to the PR that shows those OKRs

**Comment by mgoelzer** [comment]
We had some confusion on our own team when we used the reserved word `Working Group` for something more informal.  Basically just led to some misunderstandings until we changed the name.  That's the only reason I bring it up.

File: working_groups/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#### Proposal to organize into project based working groups
+Fielding, prioritizing tracking, and resourcing infra projects can be overwhelming and confusing. Additionally, many infrastructure designs and/or implementations require a close understanding of protocols and sometimes require software changes to ease deployments, maintainability, or observability. In response, the infra team would like to charge a set of project based working groups made up of members of infra team plus at least one project team developer. These working groups would meet semi regularly (frequency and format set by the WG) to communicate needs and requirements, surface and describe use cases, build an inventory of project related services, set a schedule for maintenance needs, raise security concerns, etc. The infra point of contact is charged with ensuring “meetings” happen and reports projects to the infra team as a whole for prioritizing and goal setting.
+
+The proposed working groups are as follows:
+  - `ipfs-infra`
+    - infra contact: Kyle Drake @kyledrake
+    - ipfs contact: tbd
+  - `ipfs-cluster-infra`
```
Comment (diasdavid) [inline working_groups/README.md]: Isn't this the same as IPFS Infra? Cluster gets deployed on IPFS Infra.

File: working_groups/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    - infra contact: Erin Fahy @eefahy
+    - ipfs-cluster contact: tbd
+  - [libp2p-infra](libp2p-infra.md)
+    - infra contact: Michael Burns @mburns
+    - go-libp2p contact: Cole Brown @bigs
+    - js-libp2p contact: Jake Huen @jacobheun
```
Comment (diasdavid) [inline working_groups/README.md]: What is going to be libp2p Infra?

File: working_groups/README.md (hunk 1 above)
Comment (eefahy) [reply working_groups/README.md]: Cluster has its own AWS account for running ipfs-cluster for specific use cases and therefore is a separate deploy environment with concerns of its own

File: working_groups/README.md (hunk 2 above)
Comment (eefahy) [reply working_groups/README.md]: There are also a set of bootstrap nodes (yes they are IPFS nodes at the moment) but this team, as it solidifies libp2p as a project on its own, can own the management those nodes

**Comment by diasdavid** [comment]
I believe that this proposal is mixing the need to grow 3 separate Infrastructure teams, IPFS (shared with libp2p), Filecoin and Protocol Labs and the need to assign PoC from each organization for those Infrastructure teams.

IPFS Infra should work closely with Cluster, go-ipfs, js-ipfs and also GUI, Web Browsers and all the other Working Groups.

**Comment by eefahy** [comment]
Yes, as I said, I'm using the term "working group" here very loosely here. Happy to use a different term to avoid the confusion. However, I feel like the IPFS Infra WG is in an awkward position as it has direct dependencies on other IPFS WGs for all of it's goals.

The stated responsibilities of the IPFS Infra WG are:
> Manage and maintain the IPFS HTTP Gateway.

dependency on the go-ipfs team

>Manage and maintain the IPFS Bootstrapper nodes.

dependency on the libp2p team

>Manage and maintain the IPFS pinning service, pinbot.

dependency on the cluster team

> Monitor services and hosts used by the IPFS dev teams.

dependency on all IPFS WGs

> Provide wisdom to other users that want to host IPFS nodes.

dependency on all IPFS WGs

The defined teams listed in this proposal is an attempt to create a forum for streamlined communication with a subset of the other IPFS WGs (particularly ones that have OKRs that require infra) so that the Infra WG is successful in its goals and supporting teams. Perhaps what I'm proposing here is that the IPFS Infra WG be disbanded and in its place we just assign a PL Infra team member to particular IPFS WGs so that OKRs that requires Infra support, or when the Infra team needs to make an OKR for a IPFS WG, everyone knows who to talk to for KR assignment.

**Comment by eefahy** [comment]
I've updated the proposal to charge Task Forces instead of Working Groups.

**Comment by mgoelzer** [comment]
>3 separate Infrastructure teams, IPFS (shared with libp2p)

Makes sense to share resources with IPFS, but I also want libp2p to do scale tests and publish benchmarks of pure libp2p (no IPFS involved, just using our own daemons and docker images, etc).  What I mean is that the test infra probably looks identical initially (IPTB), but over time libp2p may evolve to need more specialized configurations.

Right now we're in the early days of libp2p testing:  50k node scale is our Q4 goal [*], but next we want to start pushing go-libp2p's daemon up toward 10^5 and eventually 10^6 scale.

[*] Don't mean 50k EC instances @eefahy; many processes should run on one XL box.

File: working_groups/README.md (hunk 2 above)
Comment (mgoelzer) [reply working_groups/README.md]: Wait, this worries me, though I may be misunderstanding.  Why would the libp2p folks own management of the bootstrap peers?  Cole and Jake are software engineers; keeping the bootstrap peers running seems like a pure infra task.  What am I missing?

File: task_forces/README.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    - js-libp2p contact: Jake Huen @jacobheun
+  - `fil-infra`
+    - infra contact: George Masgras @gmasgras
+    - fil contact: Forrest Weston @frrist
+
+In addition to accomplishing project team OKRs, each task force is charged with the following set of deliverables for Q4/Q1
+ - Make a list of project and infra related repositories (for tracking state)
+ - Claim ownership of existing services as scoped by the working group
+   - devs claim ownership of the repo hygiene, CI implementations & container/manifest definitions
+   - infra claims ownership of CD, cloud infra, observability services, maintenance, & uptime of the deployed service
```
Comment (eefahy) [inline task_forces/README.md]: @mgoelzer here's what I have as to separation of responsibilities within the task force

**Comment by bigs** [comment]
✅ from me! love the idea and think it's something we could take advantage of more broadly within the org.

**Comment by gmasgras** [comment]
✔️

**Comment by hsanjuan** [comment]
Can't tick my name. Generally OK with the goal and motivation of this. I think we'll meet with infra as we need to, sometimes myself and sometimes @lanzafame depending on the topic. For a start, I'll make a list of infra-related stuff to see where infra team can help now or in the future and we can go over it in the first meeting.

**Comment by daviddias** [comment]
Just wanted to highlight the conversation happening here -- https://docs.google.com/document/d/1kX7awmrzB3Q6_sbdeWmjldhe8OzycGTdpwMsDAZBwW4/edit?disco=AAAACU_drIY --.

I would like to propose an update to the wording around roles and responsibilities in this PR. @eefahy comment on the thread above changed my understanding of this proposal, the comment was:

> In the proposal I have:
> " - devs claim ownership of the repo hygiene, CI implementations & container/manifest definitions
> - infra claims ownership of CD, cloud infra, observability services, maintenance, & uptime of the deployed service"

The other responsibility that I want to understand where it falls is the advising and supporting other groups that are also running Infrastructure with IPFS. For example: If someone is running a bunch of go-ipfs nodes in their cloud infra, is it to the go-ipfs team to educate those users on how to manage and maintain those nodes or would these be artifacts be produced by the Infra Team?

**Comment by daviddias** [comment]
Another related thought is that if we get this structure in, we could save Infra from the overhead of running multiple OKRs and centralize all into PL Infra Team OKRs

**Comment by daviddias** [comment]
*rebased master onto this branch. It was a clean rebase.

**Comment by eefahy** [comment]
>If someone is running a bunch of go-ipfs nodes in their cloud infra, is it to the go-ipfs team to educate those users on how to manage and maintain those nodes or would these be artifacts be produced by the Infra Team?

IMO I’d say that Infra could write up a “How to run IPFS* in production” chapter to IPFS documentation & be available for questions around that topic. As we move toward automation, I would love to publish a set of OSS infra utilities or publish blog posts on how we run stuff.

**Comment by eefahy** [comment]
I'm going to rewrite the project liaisons proposal. Will be tracking that work via https://github.com/protocol/infra-team/issues/20

**Pull request closed without merging**